http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/internal/impl/ApacheHttpRequestFactory.java

```diff
@@ -66,12 +64,23 @@ public HttpRequestBase create(final HttpExecuteRequest request, final ApacheHttp
      * and other AWS services, this is allowed and required. This methods replaces
      * any occurrence of "//" in the URI path with "/%2F".
      *
+     * @see SdkHttpRequest#getUri()
      * @param uri The existing URI with double slashes not sanitized for Apache.
      * @return a new String containing the modified URI
      */
     private String sanitizeUri(URI uri) {
-        String newPath = uri.getPath().replace("//", "/%2F");
-        return uri.toString().replace(uri.getPath(), newPath);
+        String path = uri.getPath();
+        if (path.contains("//")) {
+            String newPath = path.replace("//", "/%2F");
+            String port = uri.getPort() == -1 ? "" : String.valueOf(uri.getPort());
+
+            String query = uri.getQuery() == null ? "" : uri.getQuery();
+
+            return URI.create(uri.getScheme() + "://" +
+                              uri.getHost() + port + newPath + query)
+                      .toString();
+        }
```

Comment:
What about `fragment`? Are there other parts of the URI we should also be worried about?

Reply:
We don't have fragment and other parts in`SdkHttpRequest.getUri()` either.

https://github.com/aws/aws-sdk-java-v2/blob/d23f9e77c3c2bfbfb643ce8a0846dcbc5478b23c/http-client-spi/src/main/java/software/amazon/awssdk/http/SdkHttpRequest.java#L99-L120

Reply:
Hmm... okay, future bug to fix, I suppose!